### PR TITLE
fix(onboarding): hide inAppGuide open button if new onboarding enabled

### PR DESCRIPTION
**Pull request by jomifepe**
**Resolves**: DT-2234

### Description

Screenshot 2024-07-29 at 13 18 18

### Checklist

- [ ] If my changes require tests, I added them.
- [ ] If my changes affect backward compatibility, it has been discussed.
- [ ] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### How to QA [^1]

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

File: packages/slice-machine/src/features/inAppGuide/InAppGuideDialog.tsx [hunk 1]
```diff
@@ -23,8 +24,10 @@ export const InAppGuideDialog: FC = () => {
   const { isInAppGuideOpen, setIsInAppGuideOpen } = useInAppGuide();
   const inAppGuideContent = useInAppGuideContent();
 
+  const { eligible: isNewOnboardingEnable } = useOnboardingExperiment();
```
Comment (dani-mp) [inline packages/slice-machine/src/features/inAppGuide/InAppGuideDialog.tsx]:
```suggestion
  const { eligible: newOnboardingEnabled } = useOnboardingExperiment();
```

File: packages/slice-machine/src/features/inAppGuide/InAppGuideDialog.tsx [hunk 2]
```diff
@@ -23,8 +24,10 @@ export const InAppGuideDialog: FC = () => {
...
+
   const trigger =
-    isEmptyProject === false ? (
+    isEmptyProject === false && !isNewOnboardingEnable ? (
```
Comment (dani-mp) [inline packages/slice-machine/src/features/inAppGuide/InAppGuideDialog.tsx]:
```suggestion
    !isEmptyProject && !isNewOnboardingEnable ? (
```

File: packages/slice-machine/src/features/inAppGuide/InAppGuideDialog.tsx [hunk 3]
```diff
@@ -23,8 +24,10 @@ export const InAppGuideDialog: FC = () => {
   const { isInAppGuideOpen, setIsInAppGuideOpen } = useInAppGuide();
   const inAppGuideContent = useInAppGuideContent();
 
+  const { eligible: newOnboardingEnable } = useOnboardingExperiment();
```
Comment (dani-mp) [inline packages/slice-machine/src/features/inAppGuide/InAppGuideDialog.tsx]:
```suggestion
  const { eligible: newOnboardingEnabled } = useOnboardingExperiment();
```

**Review by BohdanOne — APPROVED** [review]
🎉

**Review by dani-mp — APPROVED** [review]

**Pull request merged**